src/core/Crust/Widgets/useWidgetAlignSystem.ts

```diff
@@ -14,6 +14,22 @@ export default ({ alignSystem }: { alignSystem: WidgetAlignSystem | undefined })
     alignSystem,
   );
 
+  // NOTE: This is invisible list of widget.
+  //       The reason why we use invisible list is prevent initializing cost.
+  const [invisibleWidgets, setInvisibleWidgets] = useState<string[]>([]);
+
+  const onVisible = useCallback((widgetId: string, v: boolean) => {
```

Comment:
`onVisibilityChange` I think is clearer. At first, I thought this function was dealing with only making a widget visible.